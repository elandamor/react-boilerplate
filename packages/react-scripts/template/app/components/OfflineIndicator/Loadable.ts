/**
 *
 * Asynchronously loads the component for OfflineIndicator
 *
 */

import React from 'react';

// @ts-ignore
export default React.lazy(() => import('./index'));
