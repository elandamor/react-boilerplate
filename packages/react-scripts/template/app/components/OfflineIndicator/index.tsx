import React, { SFC } from 'react';
import classNames from 'classnames';
import { WifiOff } from 'react-feather';
// Styles
import Wrapper from './styles';

/**
 * @render react
 * @name OfflineIndicator component
 * @description OfflineIndicator component.
 * @example
 * <OfflineIndicator />
 */

interface IProps {
  className?: string;
};

const OfflineIndicator: SFC<IProps> = ({ className }) => (
  <Wrapper className={classNames('', className)}>
    <WifiOff />
  </Wrapper>
);

export default OfflineIndicator;
