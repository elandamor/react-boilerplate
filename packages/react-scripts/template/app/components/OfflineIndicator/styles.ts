import styled from 'styled-components';

const Wrapper = styled.div`
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100vw;

  svg {
    color: ${({ theme }) => theme.palette.cardBorderColor};
    height: 20vw;
    max-height: 64px;
    max-width: 64px;
    width: 20vw;
  }
`;

export default Wrapper;
