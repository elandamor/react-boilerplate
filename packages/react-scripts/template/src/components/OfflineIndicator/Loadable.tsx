/**
 *
 * Asynchronously loads the component for OfflineIndicator
 *
 */

import React from 'react';
import { Loadable } from '../../utils';
import LoadingBar from '../LoadingBar';

export default Loadable(() => import('./index'), {
  fallback: <LoadingBar />,
});
