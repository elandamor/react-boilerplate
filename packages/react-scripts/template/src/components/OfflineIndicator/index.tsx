import React, { FC } from 'react';
import classNames from 'classnames';
import { WifiOff } from 'react-feather';
// Styles
import Wrapper from './styles';

interface IProps {
  className?: string;
}

/**
 * @render react
 * @name OfflineIndicator component
 * @description OfflineIndicator component.
 * @example
 * <OfflineIndicator />
 */

const OfflineIndicator: FC<IProps> = ({ className }) => (
  <Wrapper className={classNames('', className)}>
    <WifiOff />
  </Wrapper>
);

export default OfflineIndicator;
