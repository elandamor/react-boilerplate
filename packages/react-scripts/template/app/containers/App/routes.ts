// tslint:disable:object-literal-sort-keys
import { Home } from '../../pages';

import { IRouteProps } from '../../components/Routes';

const routes: IRouteProps[] = [
  {
    exact: true,
    path: '/',
    component: Home,
  },
];

export default routes;
