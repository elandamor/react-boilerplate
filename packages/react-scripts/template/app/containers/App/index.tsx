import classNames from 'classnames';
import React, { Component, Suspense } from 'react';
import Measure from 'react-measure';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
// Components
import { ErrorBoundary, LoadingBar, Routes } from '../../components';
// Contexts
import NetworkStatusProvider from '../../contexts/networkStatus.context';
// Routes
import routes from '../../routes';
// Styles
import GlobalStyles from '../../global-styles';
import Wrapper from './styles';

import { breakpoints, makeDebugger } from '../../utils';
const debug = makeDebugger('App');

const defaultTheme = {
  breakpoints: {
    small: '600px',
    medium: '1024px',
    large: '1440px',
    xLarge: '1920px',
  },
};

/* tslint:disable:object-literal-sort-keys */
const themeLight = {
  ...defaultTheme,
  isDark: false,
  palette: {
    bodyBackground: '#FAFAFA',
    brandPrimary: '#FFFFFF',
    cardBackground: '#FFFFFF',
    cardBorderColor: '#E4E6E9',
  },
};
/* tslint:enable:object-literal-sort-keys */

export interface IProps extends RouteComponentProps<any> {}

interface IState {
  bounds: {
    height: number;
    width: number;
  };
  theme: object;
}

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */
class App extends Component<IProps, IState> {
  protected componentIsMounted: boolean;
  protected previousLocation: object;

  constructor(props: IProps) {
    super(props);

    this.state = {
      bounds: {
        height: 0,
        width: 0,
      },
      theme: themeLight,
    };

    this.previousLocation = props.location;
  }

  public componentDidMount() {
    this.componentIsMounted = true;
    debug('componentIsMounted');
  }

  public componentWillUpdate(nextProps: IProps) {
    const { location } = this.props;

    if (
      nextProps.history.action !== 'POP' &&
      (!location.state || !location.state.modal)
    ) {
      this.previousLocation = this.props.location;
    }
  }

  public componentWillUnmount() {
    this.componentIsMounted = false;
  }

  public setState(nextState: any, cb?: () => void) {
    if (this.componentIsMounted) {
      super.setState(nextState, cb);
    }
  }

  public render() {
    const { location } = this.props;
    const {
      bounds: { width },
    } = this.state;

    const isModal = !!(
      location.state &&
      location.state.modal &&
      this.previousLocation !== location
    );

    return (
      <ThemeProvider theme={this.state.theme}>
        <NetworkStatusProvider>
          <Measure
            bounds
            onResize={(contentRect) => {
              this.setState({ bounds: contentRect.bounds });
            }}
          >
            {({ measureRef }) => (
              <Wrapper
                className={classNames('c-app__container', breakpoints(width))}
                // @ts-ignore
                ref={measureRef}
              >
                <GlobalStyles />
                <ErrorBoundary>
                  <Suspense fallback={<LoadingBar loading />}>
                    <Routes
                      location={isModal ? this.previousLocation : location}
                      routes={routes}
                    />
                  </Suspense>
                </ErrorBoundary>
              </Wrapper>
            )}
          </Measure>
        </NetworkStatusProvider>
      </ThemeProvider>
    );
  }
}

export default withRouter(App);
