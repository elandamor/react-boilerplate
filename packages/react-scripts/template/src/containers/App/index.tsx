import React, { Suspense } from 'react';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Normalize } from 'styled-normalize';
import get from 'lodash/get';
import merge from 'lodash/merge';
// Components
import { ErrorBoundary, Header, LoadingBar, Routes } from '@app/components';
// Contexts
import NetworkStatusProvider from '@app/contexts/networkStatus.context';
// Routes
import routes from '@app/routes';
// Styles
import { Wrapper } from './styles';

import GlobalStyles from '@app/global-styles';

import baseTheme from '@app/theme';
import { useDarkMode } from '@app/hooks';

// import { makeDebugger } from '@app/utils';
// const debug = makeDebugger('App');

export interface IAppProps extends RouteComponentProps {}

export const AppContext = React.createContext({});

/**
 * @render react
 * @name App container
 * @description The skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar).
 */

const App = () => {
  const [darkMode, setDarkMode] = useDarkMode();

  // Merge the color mode with the base theme to create a new theme object
  const getTheme = (mode: string) => merge({}, baseTheme, {
    colors: get(baseTheme.colors.modes, mode, baseTheme.colors),
    isDark: true,
  })

  const theme = darkMode ? getTheme('dark') : baseTheme;

  return (
    <ThemeProvider theme={theme}>
      <NetworkStatusProvider>
        <AppContext.Provider value={{ darkMode, setDarkMode }}>
          <Wrapper>
            <Normalize />
            <GlobalStyles />
            <ErrorBoundary>
              <Header alignItems="center" flex="none" />
            </ErrorBoundary>
            <ErrorBoundary>
              <Suspense fallback={<LoadingBar />}>
                <Routes routes={routes} />
              </Suspense>
            </ErrorBoundary>
          </Wrapper>
        </AppContext.Provider>
      </NetworkStatusProvider>
    </ThemeProvider>
  );
}

export default withRouter(App);
