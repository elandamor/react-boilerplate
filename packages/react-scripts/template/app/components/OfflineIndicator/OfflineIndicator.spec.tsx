// OfflineIndicator.spec.tsx
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import 'whatwg-fetch';

import { ThemeProvider } from 'styled-components';

import OfflineIndicator from './index';
import { themeLight } from '../../containers/App';

const theme = {
  ...themeLight,
};

describe('<OfflineIndicator />', () => {
  it('should display an offline indicator', () => {
    const component = renderer.create(
      <ThemeProvider theme={theme}>
        <OfflineIndicator />
      </ThemeProvider>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
