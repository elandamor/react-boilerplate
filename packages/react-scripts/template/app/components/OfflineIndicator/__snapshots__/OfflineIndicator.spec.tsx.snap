// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OfflineIndicator /> should display an offline indicator 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 64px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  top: 0;
  right: 0;
  width: 64px;
}

.c0 svg {
  color: #E4E6E9;
  height: 24px;
  width: 24px;
}

<div
  className="c0"
>
  <svg
    fill="none"
    height="24"
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="1"
      x2="23"
      y1="1"
      y2="23"
    />
    <path
      d="M16.72 11.06A10.94 10.94 0 0 1 19 12.55"
    />
    <path
      d="M5 12.55a10.94 10.94 0 0 1 5.17-2.39"
    />
    <path
      d="M10.71 5.05A16 16 0 0 1 22.58 9"
    />
    <path
      d="M1.42 9a15.91 15.91 0 0 1 4.7-2.88"
    />
    <path
      d="M8.53 16.11a6 6 0 0 1 6.95 0"
    />
    <line
      x1="12"
      x2="12"
      y1="20"
      y2="20"
    />
  </svg>
</div>
`;
